x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/common/utils/audit.ts

```diff
@@ -74,7 +77,8 @@ interface SiemMigrationAuditEvent {
 export class SiemMigrationAuditLogger {
   private auditLogger?: AuditLogger | null = null;
   constructor(
-    private readonly securitySolutionContextPromise: Promise<SecuritySolutionApiRequestHandlerContext>
+    private readonly securitySolutionContextPromise: Promise<SecuritySolutionApiRequestHandlerContext>,
+    private readonly migrationType: 'rules' | 'dashboards' = 'rules'
```

Comment:
Would it make more sense to make the migrationType parameter required instead of defaulting to `'rules'`?

I get that this would increase the number of changed files, but `'rules'` feels like an arbitrary default and could make it easier to forget to explicitly specify the type for other migrations. Making it required would be more explicit and safer long-term.